internal/addrs/parse_ref.go

```diff
@@ -113,6 +113,24 @@ func ParseRefFromTestingScope(traversal hcl.Traversal) (*Reference, tfdiags.Diag
 			Remaining:   remain,
 		}
 		diags = checkDiags
+	case "run":
+		if len(traversal) != 3 {
+			diags = diags.Append(&hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  "Invalid reference",
+				Detail:   `The "run" block output values must be followed by two attribute names: the resource type and the property. For eg: run.resource_type.property.`,
+				Subject:  traversal.SourceRange().Ptr(),
+			})
+			return nil, diags
+		}
+
+		remainingRefs := traversal[1:]
+		outputVariable := remainingRefs[len(remainingRefs)-1]
+		return &Reference{
+			Subject:     OutputValue{Name: outputVariable.(hcl.TraverseAttr).Name},
```

Comment:
Here we are trying to access the outputVariable from the previous "run" block only, and `outputVariable` is initialized with `remainingRefs`, which contains the output variable path (escaped first occurrence from traversal, i.e., "run")
But still I will add a check to avoid panic.